clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonApplicationProvider.groovy

```diff
@@ -36,48 +30,53 @@ import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.*
 class AmazonApplicationProvider implements ApplicationProvider {
   private final AmazonCloudProvider amazonCloudProvider
   private final Cache cacheView
-  private final ObjectMapper objectMapper
 
   @Autowired
-  AmazonApplicationProvider(AmazonCloudProvider amazonCloudProvider, Cache cacheView, @Qualifier("amazonObjectMapper") ObjectMapper objectMapper) {
+  AmazonApplicationProvider(AmazonCloudProvider amazonCloudProvider, Cache cacheView) {
     this.amazonCloudProvider = amazonCloudProvider
     this.cacheView = cacheView
-    this.objectMapper = objectMapper.enable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
   }
 
   @Override
   Set<Application> getApplications(boolean expand) {
-    def relationships = expand ? RelationshipCacheFilter.include(CLUSTERS.ns) : RelationshipCacheFilter.none()
-    Collection<CacheData> applications = cacheView.getAll(
-      APPLICATIONS.ns, cacheView.filterIdentifiers(APPLICATIONS.ns, "${amazonCloudProvider.id}:*"), relationships
-    )
-    applications.collect this.&translate
+    String allAwsGlob = "${amazonCloudProvider.id}:*"
+
+    //ignoring expand since we are deriving existence of the app by presence of server groups
+    // rather than the application cacheData which is not reliably updated or evicted
+    Map<String, Map<String, Set<String>>> appClusters = getAppClustersByAccount(allAwsGlob)
+    return appClusters.findResults {translate(it.key, appClusters) }
   }
 
   @Override
   Application getApplication(String name) {
-    translate(cacheView.get(APPLICATIONS.ns, Keys.getApplicationKey(name)))
+    name = name.toLowerCase()
+    String glob = Keys.getServerGroupKey("${name}*", "*", "*", "*")
```

Comment:
Should this be `${name}-*` ... would it match on `clouddriver` and `clouddriverproxy` otherwise?

Reply:
this is intentional - the globbing doesn't really support regex syntax. The case where it matches multiple applications is handled by grouping all the server groups in getAppClustersByAccount. There may be extra apps in there but they will not end up in the result

Reply:
(if we did `${name}-` we would require all server groups had a stack or detail)